packages/core/components/FileList/index.tsx

```diff
@@ -104,47 +104,56 @@ export default function FileList(props: FileListProps) {
                 }}
                 ref={measuredNodeRef}
             >
-                <InfiniteLoader
-                    key={fileSet.instanceId} // force a re-render whenever FileSet changes
-                    isItemLoaded={fileSet.isFileMetadataLoaded}
-                    loadMoreItems={debouncePromise<any>(
-                        fileSet.fetchFileRange,
-                        DEBOUNCE_WAIT_FOR_DATA_FETCHING
-                    )}
-                    itemCount={totalCount}
-                >
-                    {({ onItemsRendered, ref: innerRef }) => {
-                        const callbackRef = (instance: FixedSizeList | null) => {
-                            listRef.current = instance;
-
-                            // react-window-infinite-loader takes a reference to the List component instance:
-                            // https://github.com/bvaughn/react-window-infinite-loader/blob/571f6c37b692d2e01bd3b762cdc93ca7c8f7ebf3/src/InfiniteLoader.js#L103-L105
-                            if (isFunction(innerRef)) {
-                                innerRef(instance);
-                            }
-                        };
-
-                        return (
-                            <FixedSizeList
-                                itemData={{
-                                    fileSet: fileSet,
-                                    onSelect,
-                                    onContextMenu: onFileRowContextMenu,
-                                }}
-                                itemSize={rowHeight} // row height
-                                height={height} // height of the list itself; affects number of rows rendered at any given time
-                                itemCount={totalCount}
-                                onItemsRendered={onItemsRendered}
-                                outerElementType={Header}
-                                outerRef={outerRef}
-                                ref={callbackRef}
-                                width={measuredWidth}
-                            >
-                                {LazilyRenderedRow}
-                            </FixedSizeList>
-                        );
-                    }}
-                </InfiniteLoader>
+                {/* 
+                    When this component isRoot the height is measured. It takes
+                    a few milliseconds for that to happen along with a re-render, but
+                    by then the <InfiniteLoader /> used here will already have made a 
+                    request for files based on that height that it didn't need to
+                    if it had just waited until the measured height was present.
+                 */}
+                {height !== 0 && (
```

Comment:
Doesn't the change on line 67 (`|| MAX_NON_ROOT_HEIGHT`) prevent this condition from ever being `false`?

Reply:
No, at least that is not my intention. The default `|| MAX_NON_ROOT_HEIGHT` is used to give a default to the calculated height when it is not present, which at the moment is unlikely to happen, but with future work (like SWE-223) it is possible it would. Meanwhile the `height !== 0` case it to prevent rendering the `<InfiniteLoader />` before the measured height (which is different than the calculated height) is available; no default is needed for the measured height since the measurement & re-render happens within a few milliseconds..